autonomy/cli/scaffold_fsm.py

```diff
@@ -0,0 +1,769 @@
+        base_behaviour_cls_name = (
+            _try_get_base_behaviour_cls_name_from_abci_app_cls_name(abci_app_cls_name)
+        )
+        return self.BASE_BEHAVIOUR_CLS_TEMPLATE.format(
+            BaseBehaviourCls=base_behaviour_cls_name
+        )
+
+    def _get_behaviours_section(self) -> str:
+        """Get the behaviours section of the module (i.e. the list of behaviour classes)."""
+        all_behaviour_classes_str = []
+
+        abci_app_cls_name = _get_abci_app_cls_name_from_dfa(self.dfa)
+
+        # add behaviour classes
+        for abci_round_name in self.dfa.states:
+            # try to replace 'Round' suffix with 'Behaviour'
+            abci_behaviour_name = _try_get_behaviour_cls_name_from_round_cls_name(
+                abci_round_name
+            )
+            base_behaviour_cls_name = (
+                _try_get_base_behaviour_cls_name_from_abci_app_cls_name(
+                    abci_app_cls_name
+                )
+            )
+            behaviour_class_str = BehaviourFileGenerator.BEHAVIOUR_CLS_TEMPLATE.format(
+                BehaviourCls=abci_behaviour_name,
+                BaseBehaviourCls=base_behaviour_cls_name,
+            )
+            all_behaviour_classes_str.append(behaviour_class_str)
+
+        # build final content
+        return "\n".join(all_behaviour_classes_str)
+
+    def _get_behaviour_set(self) -> Set[str]:
+        """Get the set of behaviour states (excluding final states)."""
+        result: Set[str] = set()
+        for abci_round_name in self.dfa.states:
+            if abci_round_name in self.dfa.final_states:
+                continue
+            # try to replace 'Round' suffix with 'Behaviour'
+            abci_behaviour_name = _try_get_behaviour_cls_name_from_round_cls_name(
+                abci_round_name
+            )
+            result.add(abci_behaviour_name)
+        return result
+
+    def _get_round_behaviour_section(self) -> str:
+        """Get the round behaviour section of the module (i.e. the declaration of the round behaviour class)."""
+        abci_app_cls_name = _get_abci_app_cls_name_from_dfa(self.dfa)
+        round_behaviour_cls_name = (
+            _try_get_round_behaviour_cls_name_from_abci_app_cls_name(abci_app_cls_name)
+        )
+        initial_round_cls_name = self.dfa.default_start_state
+        initial_behaviour_cls_name = _try_get_behaviour_cls_name_from_round_cls_name(
+            initial_round_cls_name
+        )
+        return BehaviourFileGenerator.ROUND_BEHAVIOUR_CLS_TEMPLATE.format(
+            RoundBehaviourCls=round_behaviour_cls_name,
+            InitialBehaviourCls=initial_behaviour_cls_name,
+            AbciAppCls=abci_app_cls_name,
+            behaviours=_remove_quotes(str(self._get_behaviour_set())),
+        )
+
+
+class ModelsFileGenerator(AbstractFileGenerator):
+    """File generator for 'models.py' modules."""
+
+    FILENAME = MODELS_FILENAME
+
+    MODEL_FILE_TEMPLATE = dedent(
```

Comment:
would it be a good idea to pull these constants out and define them as constants outside of the class, such that we can use them them in tests? Or is this merely meant to setup some scaffold and then after wars one can simply edit it to one's liking?